data/rulesets.ts

```diff
@@ -862,11 +862,8 @@ export const BattleFormats: {[k: string]: FormatsData} = {
 				if (baseSpecies.otherFormes) {
 					for (const formeid of baseSpecies.otherFormes) {
 						const forme = dex.getSpecies(formeid);
-						if (!forme.battleOnly) {
-							if (!forme.forme.includes('Alola') && forme.forme !== 'Galar' && forme.baseSpecies !== 'Wormadam') {
-								types = types.concat(forme.types).concat(baseSpecies.types);
-							}
-						}
+						if (!forme.changesFrom) continue;
+						types = types.concat(forme.types).concat(baseSpecies.types);
 					}
 				}
```

Comment:
This whole block (up until `const dex =`) is implemented very messily and only works because no Pokémon has `changesFrom` and also other formes.

I'd strongly recommend this instead:

```ts
				const dex = this.dex;
				let types;
				if (species.forme || species.otherFormes) {
					const baseSpecies = dex.getSpecies(species.baseSpecies);
					const originalForme = dex.getSpecies(species.changesFrom) || species;
					types = originalForme.types;
					for (const formeid of baseSpecies.otherFormes) {
						const forme = dex.getSpecies(formeid);
						if (forme.changesFrom === originalForme.name) {
							types = types.concat(forme.types);
						}
					}
				} else {
					types = species.types;
				}

				let prevo = species.prevo;
				while (prevo) {
					const prevoSpecies = dex.getSpecies(prevo);
					types = types.concat(prevoSpecies.types);
					prevo = prevoSpecies.prevo;
				}
```